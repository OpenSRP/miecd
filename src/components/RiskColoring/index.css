.badge {
    padding: 0;
    width: 70px;
    height: 24px;
    border-radius: 10px;
    color: white;
}

.badge div {
    height: 100%;
    position: relative;
}

.badge div p {
    text-align: center;
    position: relative;
    font-size: 12px;
    font-weight: 400;
    line-height: 24px;
}
