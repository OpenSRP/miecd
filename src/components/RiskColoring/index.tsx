import { useTranslation } from 'react-i18next';
import { PREGNANCY_MODULE } from '../../constants';
import { riskCategories } from 'configs/settings';
import { keyBy } from 'lodash';
import { GREY } from 'configs/colors';
import React from 'react';
import './index.css';

interface RiskColoringProps {
    risk: string;
}

const riskColoringDefaultProps: RiskColoringProps = {
    risk: '',
};

/** Colour codes risk levels */
const RiskColoring = (props: RiskColoringProps) => {
    const { risk } = props;
    const { t } = useTranslation();

    const module = PREGNANCY_MODULE;
    const categories = riskCategories(t)[module];
    const categoriesByRisk = keyBy(categories, (item) => item.value);

    return (
        <span style={{ backgroundColor: categoriesByRisk[risk]?.color ?? GREY }} className="badge">
            <div>
                <p>{categoriesByRisk[risk]?.label ?? t('Not set')}</p>
            </div>
        </span>
    );
};

RiskColoring.defaultProps = riskColoringDefaultProps;

export default RiskColoring;
