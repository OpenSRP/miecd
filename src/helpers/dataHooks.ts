import {
    OPENSRP_SECURITY_AUTHENTICATE,
    LOCATION_FILTER_PARAM,
    RISK_CATEGORY_FILTER_PARAM,
    SMS_TYPE_FILTER_PARAM,
    SEARCH_FILTER_PARAM,
    PREGNANCY_DETECTION,
    UPDATE_PREGNANCY_DETECTION,
    NUTRITION_REGISTRATION,
    UPDATE_NUTRITION_REGISTRATION,
    DEATH_REPORT,
    UPDATE_DEATH_REPORT,
} from '../constants';
import { getQueryParams } from 'containers/LogFace/utils';
import { TFunction } from 'i18next';
import { useQuery } from 'react-query';
import { useSelector } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { OpenSRPService } from 'services/opensrp';
import { getSmsDataByFilters } from 'store/ducks/sms_events';
import { TreeNode } from './locationHierarchy/types';
import { generateJurisdictionTree } from './locationHierarchy/utils';
import { toastToError, getRiskCatFilter } from './utils';
import { LogFaceModules } from 'configs/settings';

const selectSmsData = getSmsDataByFilters();

/** hook to get this users assignment details, hierarchy, team assigned and location the team is assigned
 * @param t - the translator function
 */
export const useUserAssignment = (t: TFunction) => {
    return useQuery(
        'userAssignment',
        () => {
            const opensrpService = new OpenSRPService(OPENSRP_SECURITY_AUTHENTICATE);
            return opensrpService.read('');
        },
        {
            select: (response) => {
                return {
                    userHierarchy: generateJurisdictionTree(response.locations),
                    user: response.user,
                    team: response.team.team,
                    location: response.team.team.location,
                };
            },
            onError: (err: Error) => toastToError(err?.message ?? t('An error occurred')),
        },
    );
};

/** redux hok to select filtered sms data based on provided filters
 * @param location - route location details, to get search params
 * @param module - the module
 * @param userLocationTree - for filtering based on the user's assigned level
 */
export const useGetLogFaceFilteredSms = (
    location: RouteComponentProps['location'],
    module: LogFaceModules,
    userHierarchy?: TreeNode,
) => {
    return useSelector((state) => {
        const userLocationIdFilter = getQueryParams(location)[LOCATION_FILTER_PARAM] as string;
        const riskCategoryFilter = getQueryParams(location)[RISK_CATEGORY_FILTER_PARAM] as string;
        const smsTypeFilter = getQueryParams(location)[SMS_TYPE_FILTER_PARAM] as string;
        const searchFilter = getQueryParams(location)[SEARCH_FILTER_PARAM] as string;
        const matchesId = (node: TreeNode) => node.model.id === userLocationIdFilter;
        const userLocationNode = userHierarchy?.first(matchesId);
        let smsFilter = smsTypeFilter ? [smsTypeFilter] : undefined;
        switch (smsTypeFilter) {
            case PREGNANCY_DETECTION:
                smsFilter = [PREGNANCY_DETECTION, UPDATE_PREGNANCY_DETECTION];
                break;
            case NUTRITION_REGISTRATION:
                smsFilter = [NUTRITION_REGISTRATION, UPDATE_NUTRITION_REGISTRATION];
                break;
            case DEATH_REPORT:
                smsFilter = [DEATH_REPORT, UPDATE_DEATH_REPORT];
                break;
            default:
                break;
        }
        return selectSmsData(state, {
            locationNode: userLocationNode,
            riskCategory: getRiskCatFilter(module, riskCategoryFilter),
            smsTypes: smsFilter,
            searchFilter,
            module,
        });
    });
};
