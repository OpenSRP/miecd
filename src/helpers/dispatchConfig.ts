import { ConfigState, setAllConfigs, getAllConfigs } from '@opensrp/pkg-config';
import { OPENSRP_REST_API_BASE_URL } from 'configs/settings';
import { BACKEND_ACTIVE } from '../configs/env';
import { BACKEND_LOGIN_URL, REACT_LOGIN_URL } from '../constants';
import i18n from '../mls';

export const APP_LOGIN_URL = BACKEND_ACTIVE ? BACKEND_LOGIN_URL : REACT_LOGIN_URL;

const defaultvalues = getAllConfigs();

const configObject: ConfigState = {
    ...defaultvalues,
    appLoginURL: APP_LOGIN_URL,
    opensrpBaseURL: OPENSRP_REST_API_BASE_URL,
    i18n,
};

setAllConfigs(configObject);
