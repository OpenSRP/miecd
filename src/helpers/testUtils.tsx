import { mount, MountRendererProps } from 'enzyme';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import i18n from '../mls';
import { act } from 'react-dom/test-utils';

export const mountWithTranslations: typeof mount = (component: React.ReactNode, options?: MountRendererProps) => {
    return mount(<I18nextProvider i18n={i18n}>{component}</I18nextProvider>, options);
};

/**
 * Utility to await promises to resolve in tests
 * @param ms time in ms
 */
export async function waitForPromises() {
    await act(async () => {
        await new Promise((res) => setImmediate(res));
    });
}
