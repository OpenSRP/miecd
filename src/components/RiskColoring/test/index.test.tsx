import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { mountWithTranslations } from '../../../helpers/testUtils';
import RiskColoring from '../index';

describe('RiskColoring', () => {
    // eslint-disable-next-line jest/expect-expect
    it('must render without crashing', () => {
        shallow(<RiskColoring />);
    });

    it('must show correct Risk level depending on the string passed as a prop', () => {
        let wrapper = mountWithTranslations(<RiskColoring risk="high" />);
        expect(toJson(wrapper.find('RiskColoring'))).toMatchSnapshot('hight_risk');
        wrapper = mountWithTranslations(<RiskColoring risk="low" />);
        expect(toJson(wrapper.find('RiskColoring'))).toMatchSnapshot('low_risk');
        wrapper = mountWithTranslations(<RiskColoring risk="red" />);
        expect(toJson(wrapper.find('RiskColoring'))).toMatchSnapshot('red_alert');
        wrapper = mountWithTranslations(<RiskColoring risk="not set" />);
        expect(toJson(wrapper.find('RiskColoring'))).toMatchSnapshot('not_set');
        wrapper = mountWithTranslations(<RiskColoring />);
        expect(toJson(wrapper.find('RiskColoring'))).toMatchSnapshot('default');
    });
});
