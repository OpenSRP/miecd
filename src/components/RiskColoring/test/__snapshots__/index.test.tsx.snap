// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RiskColoring must show correct Risk level depending on the string passed as a prop: default 1`] = `
<RiskColoring
  risk=""
>
  <span
    className="badge"
    style={
      Object {
        "backgroundColor": "#6c757d",
      }
    }
  >
    <div>
      <p>
        Not set
      </p>
    </div>
  </span>
</RiskColoring>
`;

exports[`RiskColoring must show correct Risk level depending on the string passed as a prop: hight_risk 1`] = `
<RiskColoring
  risk="high"
>
  <span
    className="badge"
    style={
      Object {
        "backgroundColor": "#6c757d",
      }
    }
  >
    <div>
      <p>
        Not set
      </p>
    </div>
  </span>
</RiskColoring>
`;

exports[`RiskColoring must show correct Risk level depending on the string passed as a prop: low_risk 1`] = `
<RiskColoring
  risk="low"
>
  <span
    className="badge"
    style={
      Object {
        "backgroundColor": "#6c757d",
      }
    }
  >
    <div>
      <p>
        Not set
      </p>
    </div>
  </span>
</RiskColoring>
`;

exports[`RiskColoring must show correct Risk level depending on the string passed as a prop: not_set 1`] = `
<RiskColoring
  risk="not set"
>
  <span
    className="badge"
    style={
      Object {
        "backgroundColor": "#6c757d",
      }
    }
  >
    <div>
      <p>
        Not set
      </p>
    </div>
  </span>
</RiskColoring>
`;

exports[`RiskColoring must show correct Risk level depending on the string passed as a prop: red_alert 1`] = `
<RiskColoring
  risk="red"
>
  <span
    className="badge"
    style={
      Object {
        "backgroundColor": "#6c757d",
      }
    }
  >
    <div>
      <p>
        Not set
      </p>
    </div>
  </span>
</RiskColoring>
`;
